iohub/ngff.py

```diff
@@ -1565,6 +1565,50 @@ def positions(self) -> Generator[tuple[str, Position], None, None]:
             for _, position in well.positions():
                 yield position.zgroup.path, position
 
+    def rename_well(
+        self,
+        well,
+        old_well_path: str,
+        new_well_path: str,
+        modified={},
+        single_well=True,
+    ):
+        """Rename a well.
+
+        Parameters
+        ----------
+        old_well_path : str
+            Old name of well
+        new_well_path : str
+            New name of well
+        modified : dict, optional
+            Tracks modified wells
+        single well: bool, optional
+            True: renaming one well, False: renaming multiple wells
```

Comment:
`modified` and `single_well` shouldn't need to be arguments either. This function should only rename one well at a time - for example, your old_well_path is a string rather than a list of strings. To modify multiple wells, the user should call this function multiple times. How do you expect the user to interact with `modified`? Should this be an internal variable, e.g. `self._modified`?